drivers/media/i2c/mira016.inl

```diff
@@ -0,0 +1,6501 @@
+	ret = mira016pmic_write(pmic_client, 0x57, 0x02); // ledc ctrl1
+
+	// ret = mira016pmic_write(pmic_client, 0x4D, 0x01); // leda ctrl1
+	// ret = mira016pmic_write(pmic_client, 0x55, 0x10); // ledb ctrl7
+	ret = mira016pmic_write(pmic_client, 0x5D, 0x10); // ledc ctrl7
+	ret = mira016pmic_write(pmic_client, 0x61, 0x10); // led seq -- use this to turn on leds. abc0000- 1110000 for all leds
+
+	// uC, set atb and jtag high and ldo_en
+	ret = mira016pmic_write(uc_client, 12, 0xF7);
+	ret = mira016pmic_write(uc_client, 16, 0xF7); // ldo en:0
+	/*
+	 * In Mira016-bringup.py, write 11, 0xCF; 15: 0x30.
+	 * In mira016.py, write 11, 0x8D; 15, 0xFD.
+	 */
+	ret = mira016pmic_write(uc_client, 11, 0X8D);
+	ret = mira016pmic_write(uc_client, 15, 0xFD);
+	ret = mira016pmic_write(uc_client, 6, 1); // write
+
+	usleep_range(2000000, 2001000);
+
+	return 0;
+}
+
+static int mira016_probe(struct i2c_client *client)
+{
+	struct device *dev = &client->dev;
+	struct mira016 *mira016;
+	int ret;
+
+	printk(KERN_INFO "[MIRA016]: probing v4l2 sensor.\n");
+	printk(KERN_INFO "[MIRA016]: Driver Version 0.0.\n");
+
+	dev_err(dev, "[MIRA016] name: %s.\n", client->name);
+
+	mira016 = devm_kzalloc(&client->dev, sizeof(*mira016), GFP_KERNEL);
+	if (!mira016)
+		return -ENOMEM;
+
+	v4l2_i2c_subdev_init(&mira016->sd, client, &mira016_subdev_ops);
+
+	/* Check the hardware configuration in device tree */
+	if (mira016_check_hwcfg(dev))
+		return -EINVAL;
+
+	/* Parse device tree to check if dtoverlay has param skip-reg-upload=1 */
+	device_property_read_u32(dev, "skip-reg-upload", &mira016->skip_reg_upload);
+	printk(KERN_INFO "[MIRA016]: skip-reg-upload %d.\n", mira016->skip_reg_upload);
+	/* Set default TBD I2C device address to LED I2C Address*/
+	mira016->tbd_client_i2c_addr = MIRA016LED_I2C_ADDR;
+	printk(KERN_INFO "[MIRA016]: User defined I2C device address defaults to LED driver I2C address 0x%X.\n", mira016->tbd_client_i2c_addr);
+
+	/* Get system clock (xclk) */
+	mira016->xclk = devm_clk_get(dev, NULL);
+	if (IS_ERR(mira016->xclk))
+	{
+		dev_err(dev, "failed to get xclk\n");
+		return PTR_ERR(mira016->xclk);
+	}
+
+	mira016->xclk_freq = clk_get_rate(mira016->xclk);
+	if (mira016->xclk_freq != MIRA016_SUPPORTED_XCLK_FREQ)
+	{
+		dev_err(dev, "xclk frequency not supported: %d Hz\n",
+				mira016->xclk_freq);
+		return -EINVAL;
+	}
+
+	ret = mira016_get_regulators(mira016);
+	if (ret)
+	{
+		dev_err(dev, "failed to get regulators\n");
+		return ret;
+	}
+
+	{
+		printk(KERN_INFO "[MIRA016]: Init PMIC and uC and led driver.\n");
+		mira016->pmic_client = i2c_new_dummy_device(client->adapter,
+													MIRA016PMIC_I2C_ADDR);
+		if (IS_ERR(mira016->pmic_client))
+			return PTR_ERR(mira016->pmic_client);
+		mira016->uc_client = i2c_new_dummy_device(client->adapter,
+												  MIRA016UC_I2C_ADDR);
+		if (IS_ERR(mira016->uc_client))
+			return PTR_ERR(mira016->uc_client);
+		mira016->led_client = i2c_new_dummy_device(client->adapter,
```

Comment:
Other than a couple of init commands, what does the kernel do with this LED driver?